i18n-msg.html

```diff
@@ -64,166 +65,202 @@
+            lang: null,
+            url: 'locales',
+            locales: {}
+        };
+
+        var instances = [];
+
+        var fetching_ = false;
+        /* If if there's an outstanding XHR fetching a locale file. */
+
+        var numInstancesUpdated_ = 0;
+
+        Polymer({
+            is: 'i18n-msg',
+
+            properties: {
+                /**
+                 * The message id (key) for this message.
+                 *
+                 * @attribute msgid
+                 * @type string
+                 * @default null
+                 */
+                msgid: {
+                    type: String,
+                    value: null
+                },
+
+                /**
+                 * The set language being used.
+                 *
+                 * @property language
+                 * @type string
+                 * @default null
+                 * @readonly
+                 */
+                language: {
+                    type: String,
+                    value: null,
+                    observer: 'languageChanged'
+                }
+            },
+
+            /**
+             * The folder name where the localized `<lang>.json` files are located.
+             *
+             * @attribute messagesUrl
+             * @type string
+             * @default 'locales'
+             */
+            messagesUrl: 'locales',
+
+            /**
+             * An object containing the set of known language locales that have been loaded.
+             *
+             * @property locales
+             * @type object
+             * @default {}
+             */
+            locales: I18nMsg.locales, // static
+
+            ready: function () {
+                this.language = I18nMsg.lang;
+                this.messagesUrl = I18nMsg.url;
+
+                // Have instances observe window.I18nMsg.lang changes
+                // and go fetch the localized messages.json file.
+                var observerLang = new PathObserver(window.I18nMsg, 'lang');
+                observerLang.open(function (newValue, oldValue) {
+                    numInstancesUpdated_ = 0;
+                    this.language = newValue;
+                }.bind(this));
+
+                // Have instances observe window.I18nMsg.lang changes
+                // and go fetch the localized messages.json file.
+                var observerUrl = new PathObserver(window.I18nMsg, 'url');
+                observerUrl.open(function (newValue, oldValue) {
+                    numInstancesUpdated_ = 0;
+                    this.messagesUrl = newValue;
+
+                    if (!this.language || fetching_) {
+                        return;
+                    }
+
+                    this.fetchLanguage();
+                }.bind(this));
+
+                instances.push(this);
+            },
+
+            attached: function () {
+                if (!this.msgid) {
+                    this.msgid = this.innerHTML;
```

Comment:
The intention was that if a msgid isn't found or ommitted, the `textContent` is used as fallback text. I think we should keep those semantics. Please remove.